### PR TITLE
feat(query): count_values function support

### PR DESCRIPTION
**Pull request by TanviBhavsar**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

Support for Prometheus count_values function

### CODE REVIEW

File: query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala [hunk 1]
```diff
@@ -932,3 +933,116 @@ class QuantileRowAggregator(q: Double) extends RowAggregator {
     ResultSchema(Array(reductionSchema.columns(0), ColumnInfo("value", ColumnType.DoubleColumn)), 1)
   }
 }
+
+class CountValuesRowAggregator(label: String, limit: Int = 1000) extends RowAggregator {
```
[the remark this answers is not in the document]
Comment (TanviBhavsar) [reply query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala]: Created filodb.query.exec.aggregator package and moved each aggregator to separate file

File: query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala [hunk 2]
```diff
@@ -932,3 +933,116 @@ class QuantileRowAggregator(q: Double) extends RowAggregator {
...
+    map(item.getDouble(1)) = 1
+    serialize(map)
+    mapInto.setBlob(1, arr, UnsafeUtils.arayOffset, map.size * 12)
+    mapInto
+  }
+
+  def reduceAggregate(acc: CountValuesHolder, aggRes: RowReader): CountValuesHolder = {
+    acc.timestamp = aggRes.getLong(0)
+    val aggMap = deserialize(aggRes.getBlobBase(1), aggRes.getBlobNumBytes(1),
+      aggRes.getBlobOffset(1))
```
[the remark this answers is not in the document]
Comment (TanviBhavsar) [reply query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala]: Good suggestion! Made the change

**Comment by TanviBhavsar** [comment]
Thanks Vish! I have addressed your review comments

File: query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala [hunk 3]
```diff
@@ -934,72 +935,72 @@ class QuantileRowAggregator(q: Double) extends RowAggregator {
   }
 }
 
+/**
+  * Map: Every sample is mapped to map with value as key and 1 as value
```
Comment (vishramachandran) [inline query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala]: "value as key and 1 as value" sounds funny :D 
I get you, but consider rephrasing for others.

File: query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala [hunk 4]
```diff
@@ -934,72 +935,72 @@ class QuantileRowAggregator(q: Double) extends RowAggregator {
...
 class CountValuesRowAggregator(label: String, limit: Int = 1000) extends RowAggregator {
 
-  var arr = new Array[Byte](500)
+  var serializedMap = new Array[Byte](500)
+  val rowMap = debox.Map[Double, Int]()
   class CountValuesHolder(var timestamp: Long = 0L) extends AggregateHolder {
     val frequencyMap = debox.Map[Double, Int]()
     val row =  new CountValuesTransientRow
     def toRowReader: MutableRowReader = {
       val size = frequencyMap.size * 12
```
Comment (vishramachandran) [inline query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala]: place this size calculation formula next to the ser/deser logic.

File: query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala [hunk 5]
```diff
@@ -934,72 +935,72 @@ class QuantileRowAggregator(q: Double) extends RowAggregator {
...
-      if (arr.length < size) arr = new Array[Byte](size) // Create bigger array
-      serialize(frequencyMap)
+      if (serializedMap.length < size) serializedMap = new Array[Byte](size) // Create bigger array
+      CountValuesTransientRow.serialize(frequencyMap, serializedMap)
```
Comment (vishramachandran) [inline query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala]: It appears that `object CountValuesTransientRow` can be renamed to `object CountValuesSerDeser`

File: query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala (hunk 4 above)
Comment (TanviBhavsar) [reply query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala]: Done

**Comment by TanviBhavsar** [comment]
Thanks Vish! Addressed comments and fixed PRB

**Pull request merged**